server/src/main/java/org/opensearch/search/internal/SearchContext.java

```diff
@@ -399,15 +399,15 @@ public final void assignRescoreDocIds(RescoreDocIds rescoreDocIds) {
     /**
      * Returns concurrent segment search status for the search context
      */
-    public boolean isConcurrentSegmentSearchEnabled() {
+    public boolean shouldUseConcurrentSearch() {
```

Comment:
Can we make this abstract instead ?

Reply:
Going to leave this as concrete for now since SearchContext has a bunch of anonymous implementations in various test classes. We can revisit this later but for the purpose of this PR this should be fine.